source/frontends/common2/CMakeLists.txt

```diff
@@ -17,37 +13,63 @@ set(HEADER_FILES
   commonframe.h
   commoncontext.h
   controllerdoublepress.h
-  gnuframe.h
-  fileregistry.h
-  ptreeregistry.h
-  programoptions.h
   utils.h
   timer.h
   speed.h
   )
 
+if (BUILD_APPLEN OR BUILD_QAPPLE OR BUILD_SA2)
+  set(SOURCE_FILES ${SOURCE_FILES}
```

Comment:
I will handle this in  a different way. The content of the libraries should not depend on what one is building. Better to split the libraries. Leave it with me.